recovery.go

```diff
@@ -0,0 +1,70 @@
+package raft
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// PeersJSONRecovery is a recovery manager that reads the old-style peers.json
+// file. It does not support server IDs, so it should only be used along with
+// ProtocolVersion 0. Upon disarming, it'll delete the peers.json file.
+type PeersJSONRecovery struct {
+	// path is the full path to the peers.json file.
+	path string
+
+	// Configuration is the override configuration.
+	Configuration Configuration
+}
+
+// NewPeersJSONRecovery takes the given base directory and parses the peers.json
+// file at that location, returning a recovery manager to apply this
+// configuration at startup. If there's no recovery defined then this will return
+// nil, which is a valid thing to pass to NewRaft().
+func NewPeersJSONRecovery(base string) (*PeersJSONRecovery, error) {
```

Comment:
Good catch - I originally had this horrible thing that was passed in to `NewRaft()` that waited for things to replicate, etc. This is a leftover from that so I'll rename this and remove the `Disarm()` shiz.